causalml/inference/meta/base.py

```diff
@@ -38,6 +45,57 @@ def bootstrap(self, X, treatment, y, p=None, size=10000):
         self.fit(X=X_b, treatment=treatment_b, y=y_b, p=p_b)
         return self.predict(X=X, p=p)
 
+    @staticmethod
+    def _format_p(p, t_groups):
+        """Format propensity scores into a dictionary of {treatment group: propensity scores}.
+
+        Args:
+            p (np.ndarray, pd.Series, or dict): propensity scores
+            t_groups (list): treatment group names.
+
+        Returns:
+            dict of {treatment group: propensity scores}
+        """
+        check_p_conditions(p, t_groups)
+
+        if isinstance(p, (np.ndarray, pd.Series)):
+            treatment_name = t_groups[0]
+            p = {treatment_name: convert_pd_to_np(p)}
+        elif isinstance(p, dict):
+            p = {treatment_name: convert_pd_to_np(_p) for treatment_name, _p in p.items()}
+
+        return p
+
+    def _set_propensity_models(self, X, treatment, y):
+        """Set self.propensity and self.propensity_models.
+
+        It trains propensity models for all treatment groups, save them in self.propensity_models, and
+        save propensity scores in self.propensity in dictionaries with treatment groups as keys.
+
+        It will use self.model_p if available to train propensity models. Otherwise, it will use a default
+        PropensityModel (i.e. ElasticNetPropensityModel).
+
+        Args:
+            X (np.matrix or np.array or pd.Dataframe): a feature matrix
+            treatment (np.array or pd.Series): a treatment vector
+            y (np.array or pd.Series): an outcome vector
+        """
+        logger.info('Generating propensity score')
+        p = dict()
+        p_model = dict()
+        for group in self.t_groups:
+            mask = (treatment == group) | (treatment == self.control_name)
+            treatment_filt = treatment[mask]
+            X_filt = X[mask]
+            w_filt = (treatment_filt == group).astype(int)
+            w = (treatment == group).astype(int)
+            propensity_model = self.model_p if hasattr(self, 'model_p') else None
+            p[group], p_model[group] = compute_propensity_score(X=X_filt, treatment=w_filt,
```

Comment:
Hi @jeongyoonlee a quick question that might not directly be related to this PR. Why do we need two `treatment` and `treatment_pred` here? Aren't those the same thing based on here: https://github.com/uber/causalml/blob/master/causalml/propensity.py#L189?

Reply:
In this call to `compute_propensity_score()`, `treatment` is only for the filtered samples with `treatment` belonging to either the control group or treatment group of interest while `treatment_pred` is for the entire samples. These two will be the same for a single treatment, but different for the multiple treatments.

Reply:
Gotcha I see. Sorry a follow-up question here, why do we need to calibrate it with the entire dataset? I am asking because we are only using treatment group X + control when running the model in https://github.com/uber/causalml/blob/master/causalml/inference/meta/rlearner.py#L111-L118.